The server crashed after executing SQL statement with "where spID>2 AND userID <2 || userID >=2 OR userID < 2"

<!-- Please describe your issue in English. -->

#### Can be reproduced?
Yes.

#### Steps:
Run SQL below:
select * from t1 where spID>2 AND userID <2 || userID >=2 OR userID < 2;

#### Expected behavior:
Expect that the SQL can be parsed and return correct records; if the SQL statement isn't supported, then a user-friendly error message should be issued. 

#### Actual behavior:
The server crashed immediately after executing the SQL statement.
Please see server log below:

2021/09/17 11:38:44.638201 +0800 INFO cmd 3
2021/09/17 11:38:44.638275 +0800 INFO query:select * from t1 where spID>2 AND userID <2 || use...
2021/09/17 11:38:44.648156 +0800 INFO connection id 3 , the time of handling the request 9.957208ms
2021/09/17 11:39:01.996338 +0800 INFO cmd 3
2021/09/17 11:39:01.996523 +0800 INFO query:show databases
2021/09/17 11:39:01.996875 +0800 INFO time of Exec.Compile : 22.875µs
+++++++++
:10 
+++++++++
2021/09/17 11:39:01.998021 +0800 INFO goid 20764 

2021/09/17 11:39:01.998202 +0800 INFO time of getDataFromPipeline : 142µs 
2021/09/17 11:39:01.998241 +0800 INFO time of Exec.Run : 1.218ms
2021/09/17 11:39:01.998297 +0800 INFO connection id 3 , the time of handling the request 1.980417ms
2021/09/17 11:39:07.083571 +0800 INFO cmd 3
2021/09/17 11:39:07.083726 +0800 INFO query:SELECT DATABASE()
2021/09/17 11:39:07.086720 +0800 INFO connection id 3 , the time of handling the request 3.1895ms
2021/09/17 11:39:07.086824 +0800 INFO cmd 2
2021/09/17 11:39:07.088868 +0800 INFO User dump change database from [] to [test01]

2021/09/17 11:39:07.089029 +0800 INFO connection id 3 , the time of handling the request 2.201333ms
2021/09/17 11:39:07.090526 +0800 INFO cmd 3
2021/09/17 11:39:07.090570 +0800 INFO query:show databases
2021/09/17 11:39:07.090650 +0800 INFO time of Exec.Compile : 9.541µs
+++++++++
:10 
+++++++++
2021/09/17 11:39:07.091124 +0800 INFO goid 20728 

2021/09/17 11:39:07.091175 +0800 INFO time of getDataFromPipeline : 30.208µs 
2021/09/17 11:39:07.091188 +0800 INFO time of Exec.Run : 476.042µs
2021/09/17 11:39:07.091212 +0800 INFO connection id 3 , the time of handling the request 692.667µs
2021/09/17 11:39:07.091416 +0800 INFO cmd 3
2021/09/17 11:39:07.091449 +0800 INFO query:show tables
2021/09/17 11:39:07.092520 +0800 INFO time of Exec.Compile : 693.666µs
+++++++++
:9 
+++++++++
2021/09/17 11:39:07.094406 +0800 INFO goid 20729 

2021/09/17 11:39:07.094499 +0800 INFO time of getDataFromPipeline : 61.708µs 
2021/09/17 11:39:07.094533 +0800 INFO time of Exec.Run : 1.93425ms
2021/09/17 11:39:07.094570 +0800 INFO connection id 3 , the time of handling the request 3.1505ms
2021/09/17 11:39:07.094684 +0800 INFO cmd 4
2021/09/17 11:39:07.094766 +0800 INFO connection id 3 , the time of handling the request 81.75µs
2021/09/17 11:39:07.094870 +0800 INFO cmd 4
2021/09/17 11:39:07.094930 +0800 INFO connection id 3 , the time of handling the request 60.959µs
2021/09/17 11:39:11.874649 +0800 INFO cmd 3
2021/09/17 11:39:11.874860 +0800 INFO query:select * from t1 where spID>2 AND userID <2 || use...
2021/09/17 11:39:11.892509 +0800 INFO time of Exec.Compile : 17.07375ms
+++++++++
		:0 π(spID,userID,score) -> σ(spID > 2 and userID < 2 or userID >= 2 or userID < 2) -> π(spID,userID,score) -> =>
	:2 merge -> =>
:1 merge -> sql output
+++++++++
2021/09/17 11:39:11.896870 +0800 INFO session connected {"session-id": 1, "addr": "127.0.0.1:61631"}
panic: runtime error: index out of range [7] with length 7

goroutine 20867 [running]:
matrixone/pkg/vectorize/shuffle.i32ShufflePure({0xc00322c414, 0x1c, 0x203000}, {0xc0029370c0, 0x0, 0x110000000000}, {0xc002937040, 0x0, 0x0})
	/Users/anixjj/Workspace/matrixone/pkg/vectorize/shuffle/shuffle.go:147 +0xb1
matrixone/pkg/vectorize/shuffle.I32Shuffle(...)
	/Users/anixjj/Workspace/matrixone/pkg/vectorize/shuffle/shuffle.go:78
matrixone/pkg/container/vector.(*Vector).Shuffle(0xc00322e000, {0xc002937040, 0x8, 0x8}, 0xc001febb90)
	/Users/anixjj/Workspace/matrixone/pkg/container/vector/vector.go:535 +0x579
matrixone/pkg/sql/colexec/restrict.Call(0xc0023baa00, {0x5499620, 0xc00315a330})
	/Users/anixjj/Workspace/matrixone/pkg/sql/colexec/restrict/restrict.go:95 +0x371
matrixone/pkg/vm.Run({0xc001c3ea80, 0x4, 0x3}, 0xc003226168)
	/Users/anixjj/Workspace/matrixone/pkg/vm/vm.go:49 +0x94
matrixone/pkg/vm/pipeline.(*Pipeline).Run(0xc003153f00, {0xc002b5bfb0, 0x1, 0x1}, 0xc0023baa00)
	/Users/anixjj/Workspace/matrixone/pkg/vm/pipeline/pipeline.go:93 +0x776
matrixone/pkg/sql/compile.(*Scope).Run(0xc003142c00, {0x5a86100, 0xc00079e228})
	/Users/anixjj/Workspace/matrixone/pkg/sql/compile/scope.go:77 +0x699
matrixone/pkg/sql/compile.(*Scope).MergeRun.func1(0x0)
	/Users/anixjj/Workspace/matrixone/pkg/sql/compile/scope.go:92 +0x33
created by matrixone/pkg/sql/compile.(*Scope).MergeRun
	/Users/anixjj/Workspace/matrixone/pkg/sql/compile/scope.go:91 +0x412

#### Environment:
- Version or commit-id (e.g. v0.1.0 or 8b23a93): commit c499bfc24dfac0786dd4e0d7e0356a9142d3a754
- Hardware parameters:
- OS type: macOS
- Others:

#### Configuration file:


#### Additional context:
- Error message from client:
- Server log:
- Other information:

--- a/pkg/sql/colexec/extend/overload/and.go
+++ b/pkg/sql/colexec/extend/overload/and.go
@@ -39,6 +39,7 @@ func init() {
 					if rv.Ref == 0 {
 						register.Put(proc, rv)
 					}
+					lv.Col = lvs
 					return lv, nil
 				}
 				if len(rvs) >= len(lvs) && (rv.Ref == 1 || rv.Ref == 0) {
@@ -48,6 +49,7 @@ func init() {
 					if lv.Ref == 0 {
 						register.Put(proc, lv)
 					}
+					rv.Col = rvs
 					return rv, nil
 				}
 				n := len(lvs)

